File: GeneratedEntityReaderClasses/Components/ElectricityReceiverComponent.cs
namespace GeneratedNoitaClasses;
public class ElectricityReceiverComponent : NoitaComponentBase,  iNoitaType<ElectricityReceiverComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaInt offset_x { get; set; } = new();
	public NoitaInt offset_y { get; set; } = new();
	public NoitaInt radius { get; set; } = new();
	public NoitaInt active_time_frames { get; set; } = new();
	public NoitaInt switch_on_msg_interval_frames { get; set; } = new();
	public NoitaInt electrified_msg_interval_frames { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		offset_x.Read(s);
		offset_y.Read(s);
		radius.Read(s);
		active_time_frames.Read(s);
		switch_on_msg_interval_frames.Read(s);
		electrified_msg_interval_frames.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		offset_x.Write(s);
		offset_y.Write(s);
		radius.Write(s);
		active_time_frames.Write(s);
		switch_on_msg_interval_frames.Write(s);
		electrified_msg_interval_frames.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/ElectricityReceiverComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class ElectricityReceiverComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    offset_x:NoitaInt = field(default_factory=NoitaInt)
    offset_y:NoitaInt = field(default_factory=NoitaInt)
    radius:NoitaInt = field(default_factory=NoitaInt)
    active_time_frames:NoitaInt = field(default_factory=NoitaInt)
    switch_on_msg_interval_frames:NoitaInt = field(default_factory=NoitaInt)
    electrified_msg_interval_frames:NoitaInt = field(default_factory=NoitaInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.offset_x.Read(s)
        self.offset_y.Read(s)
        self.radius.Read(s)
        self.active_time_frames.Read(s)
        self.switch_on_msg_interval_frames.Read(s)
        self.electrified_msg_interval_frames.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.offset_x.Write(s)
        self.offset_y.Write(s)
        self.radius.Write(s)
        self.active_time_frames.Write(s)
        self.switch_on_msg_interval_frames.Write(s)
        self.electrified_msg_interval_frames.Write(s)

